fedorable.sh

```
@@ -402,223 +241,160 @@ task_autoremove() {
 
-    # Try grubby first, as it's often preferred/safer on Fedora
+    # Try grubby first, then fall back to grub2-mkconfig if needed
     if check_command grubby; then
-        log_msg "Attempting GRUB update using grubby..."
-        execute_if_not_dry_run "grubby --update-kernel=ALL" "Update kernel entries via grubby"
-        if check_status "Update kernel entries via grubby"; then
-            grub_updated=1
-            # Check if default kernel was updated
+        run_cmd "grubby --update-kernel=ALL" "Update kernel entries via grubby"
+        if [[ $? -eq 0 ]]; then
              local default_kernel=$(grubby --default-kernel)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)